pkg/data/set.go

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package data
+
```

Comment:
Just to keep in mind: I think this refactor (data transformations) should be done in a different PR if we follow https://github.com/okteto/okteto/blob/master/docs/code-review.md#pr-scope. As a reviewer it becomes more complicated for me to focus on what essentially needs to be changed to close the issue.

Reply:
You are right. I will try to rebase later today to reduce the scope and open a follow up PR

Reply:
@AdrianPedriza @jLopezbarb I've reset this to the bare minimum changes in behavior that are needed to fix the issue and created a [follow up ](https://github.com/okteto/okteto/pull/3077)with the refactors